core/activity/src/provider/service.rs

```diff
@@ -350,31 +406,41 @@ async fn monitor_activity(
                 break;
             }
 
-            let dt = (Utc::now().timestamp() - usage.timestamp) as f64;
-            if dt > limit_s {
-                log::warn!("activity {} inactive for {}s, destroying", activity_id, dt);
-                enqueue_destroy_evt(
-                    db,
-                    tracker.clone(),
-                    &activity_id,
-                    provider_id,
-                    app_session_id.clone(),
-                )
-                .await;
+            let duration = (Utc::now().timestamp() - usage.timestamp) as f64;
+
+            if duration > unresponsive_limit || duration > inactivity_limit {
```

Comment:
It would be more clear if we had only `duration > unresponsive_limit` check.
The second check should be always true since we should have `unresponsive_limit < inactivity_limit`.
If the relation is opposite that the whole code makes no sense, because we make activity responsive again, after we already added ActivityDestroy event

Reply:
`unresponsive_limit` and `inactivity_limit` come from env variables, and I have found no code making sure that `unresponsive_limit < inactivity_limit`, which is why I wrote it this way.

> If the relation is opposite that the whole code makes no sense, because we make activity responsive again, after we already added ActivityDestroy event

Where it would happen?

Reply:
Changed.